packages/flutter_tools/lib/src/commands/assemble.dart

```diff
@@ -116,6 +116,7 @@ class AssembleCommand extends FlutterCommand {
         'files will be written. Must be either absolute or relative from the '
         'root of the current Flutter project.',
     );
+    usesFlavorOption();
```

Comment:
So this adds the --flavor command line flag, but you're passing the flavor with -D option which is separate.